kickstarter/android-oss
# NT-1747: Identify users on segment

# 📲 What

- Every time a user logs in or sign up we need to identify that user on segment

# 🤔 Why
- Generation of Funnels in Segment requires to Identify Users

# 🛠 How

- We listen to the currentUser observable on the base `Trackingclient` and call `identify` method.

# 👀 See
**- Debug Logcat line for identify call**
<img width="969" alt="Screen Shot 2021-02-02 at 9 02 29 AM" src="https://user-images.githubusercontent.com/4083656/106635269-7ecd7c00-6535-11eb-993f-2867f18e3668.png">

**- Android staging source** 
![Screen Shot 2021-02-02 at 8 57 36 AM](https://user-images.githubusercontent.com/4083656/106634558-bd166b80-6534-11eb-86d7-fdc0803917f9.png)

|  |  |

# 📋 QA
- Check on the android staging source that and identify call is registered when you log in in the app
- Check on the android staging source that and identify call is registered when you log in with an user with tfa challenge in the app
- Check on the android staging source that and identify call is registered when sign up in in the app
- Check on the android staging source that and identify call is registered when sign up with facebook in in the app
- Check no duplicated calls in place for the same user 

# Story 📖

[NT-1747: Identify users in segment](https://kickstarter.atlassian.net/browse/NT-1747)
